### PR TITLE
Fix get ceph pool extras bug

### LINKED ISSUE #646: osdsdb could not get the information from /etc/opensds/driver/ceph.yaml
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug

**What happened**:

I found that osdsdb could not get the information from `/etc/opensds/driver/ceph.yaml` when I tried OpenSDS v0.5.1 and v0.5.0 .

My  `/etc/opensds/driver/ceph.yaml` :

```
configFile: /etc/ceph/ceph.conf
pool:
  rbd: # change pool name same to ceph pool, but don't change it if you choose lvm backend
    storageType: block
    availabilityZone: default
    extras:
      dataStorage:
        provisioningPolicy: Thin
        isSpaceEfficient: true
      ioConnectivity:
        accessProtocol: rbd
        maxIOPS: 6000000
        maxBWS: 500
      advanced:
        diskType: SSD
        latency: 5ms
```

But `etcdctl` showed that:

```
root@sds-con:/opt/opensds-hotpot-linux-amd64# ../etcd/etcdctl get v1beta --prefix
v1beta/docks/30e5f409-164e-58e6-8aab-148377438abd
{"id":"30e5f409-164e-58e6-8aab-148377438abd","createdAt":"2019-04-09T18:02:39","updatedAt":"","type":"provisioner","name":"ceph backend","description":"This is a ceph backend service","endpoint":"10.202.5.28:50050","nodeId":"ceph-node1","driverName":"ceph","metadata":{"HostReplicationDriver":"drbd"}}
v1beta/pools/0517f561-85b3-5f6a-a38d-8b5a08bff7df
{"id":"0517f561-85b3-5f6a-a38d-8b5a08bff7df","createdAt":"2019-04-09T18:02:39","updatedAt":"","name":"rbd","dockId":"30e5f409-164e-58e6-8aab-148377438abd","availabilityZone":"default","totalCapacity":89,"freeCapacity":89,"storageType":"block","extras":{"dataStorage":{"isSpaceEfficient":false},"ioConnectivity":{},"dataProtection":{"isIsolated":false}},"replicationType":"HostBased","replicationDriverName":"drbd"}
```

As we have seen，The value of the `extras` field is empty.This will cause the `accessProtocol` 
 field to become `iscsi`.

**What you expected to happen**:

osdsdb could  get the information from `/etc/opensds/driver/ceph.yaml` correctly.
 
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Hotpot(release/branch) version:v0.5.0
- OS (e.g. from /etc/os-release):debian 9
- Kernel (e.g. `uname -a`):Linux sds-con 4.9.0-8-amd64 #1 SMP Debian 4.9.110-3+deb9u6 (2018-10-08) x86_64 GNU/Linux
- Install tools:Manual
- Others:

### PR DESCRIPTION
**Pull request by imquanquan**
**What this PR does / why we need it**:

- Fix the get ceph pool extras bug in osdsdock v0.5.1 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #646 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by leonwanghui** [comment]
@xxwjj PTAL

**Comment by xxwjj** [comment]
@imquanquan These code you commited is useless, and has beend removed  long time ago. Can you tell why you re-add these code.

**Comment by imquanquan** [comment]
Hi, @xxwjj you optimize some codes in #601 ,but removed the extras field of pools in ListPools method

https://github.com/opensds/opensds/blob/b9cd3fe65a676e8b6fc1d507bcacfb8f1b242618/contrib/drivers/ceph/ceph.go#L435-L450

The ListPools method is called by the Discover method.When the extras field is empty, etcd will use the default value.This will cause the accessProtocol field  in pool extras to become iscsi,the volumes are not available in Kubernetes.

So,I think thank re-add your code could fix this issue,Thanks

File: contrib/drivers/ceph/ceph.go [hunk 1]
```diff
@@ -440,6 +498,7 @@ func (d *Driver) ListPools() ([]*model.StoragePoolSpec, error) {
 			TotalCapacity:    (p.Stats.BytesUsed + p.Stats.MaxAvail) >> sizeShiftBit,
 			FreeCapacity:     p.Stats.MaxAvail >> sizeShiftBit,
 			StorageType:      d.conf.Pool[p.Name].StorageType,
+			Extras:           extras,
```
Comment (xxwjj) [inline contrib/drivers/ceph/ceph.go]: @imquanquan thanks,  that's a bug. To fixed the bug ,you just need to s/extras/d.conf.Pool[p.Name].Extras, and remove all other codes. propertes in extras.Advanced  is not useless right now.

Comment (imquanquan) [reply contrib/drivers/ceph/ceph.go]: Done.

**Comment by leonwanghui** [comment]
Good job @imquanquan!

**Pull request merged**